bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/internal/QueryHandler.java

```diff
@@ -0,0 +1,274 @@
+import org.openhab.binding.dbquery.internal.domain.ResultValue;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages query thing, handling it's commands and updating it's channels
+ *
+ * @author Joan Pujol - Initial contribution
+ */
+@NonNullByDefault
+public class QueryHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(QueryHandler.class);
+    // Relax nullable rules as config can be only null when not initialized
+    private @NonNullByDefault({}) QueryConfiguration config;
+    private @NonNullByDefault({}) QueryResultExtractor queryResultExtractor;
+
+    private @Nullable ScheduledFuture<?> scheduledQueryExecutionInterval;
+    private @Nullable QueryResultChannelUpdater queryResultChannelUpdater;
+    private Database database = Database.EMPTY;
+    private final DBQueryJSONEncoder jsonEncoder = new DBQueryJSONEncoder();
+
+    private @Nullable QueryExecution currentQueryExecution;
+    private QueryResult lastQueryResult = QueryResult.NO_RESULT;
+
+    public QueryHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(QueryConfiguration.class);
+        queryResultExtractor = new QueryResultExtractor(config);
+
+        initQueryResultChannelUpdater();
+        updateStateWithParentBridgeStatus();
+    }
+
+    private void initQueryResultChannelUpdater() {
+        ChannelStateUpdater channelStateUpdater = (channel, state) -> updateState(channel.getUID(), state);
+        queryResultChannelUpdater = new QueryResultChannelUpdater(channelStateUpdater, this::getResultChannels2Update);
+    }
+
+    private void scheduleQueryExecutionIntervalIfNeeded() {
+        int interval = config.getInterval();
+        if (interval != QueryConfiguration.NO_INTERVAL && scheduledQueryExecutionInterval == null) {
+            logger.trace("Scheduling query execution every {} seconds for {}", interval, getQueryIdentifier());
+            scheduledQueryExecutionInterval = Executors.newSingleThreadScheduledExecutor()
+                    .scheduleAtFixedRate(this::executeQuery, 0, interval, TimeUnit.SECONDS);
```

Comment:
Ok on framework scheduler, but here, the scheduleAtFixedRate is "necessary" because it's the documented and intended behavior.
Possibly an improvement can be to allow the user to setup the fixedDelay behaviour in config.

Reply:
If binding documentation talks about refresh intervals, they use `scheduleWithFixedDelay`. Usually you don't notice a difference and it makes it a lot easier for the scheduler.